frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,255 @@
+  const [pvcSC, setPVCStorageClass] = React.useState<string>(null);
+  const [progress, setProgress] = React.useState(true);
+  const [driver, setDriver] = React.useState<string>(null);
+  const [pvcData, setPVCData] = React.useState<PersistentVolumeClaimKind>(null);
+  const namespace = getNamespace(resource);
+
+  React.useEffect(() => {
+    Promise.all([
+      k8sGet(
+        PersistentVolumeClaimModel,
+        resource?.spec?.source?.persistentVolumeClaimName,
+        namespace,
+      ).catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching objects', e); // Not important to show user
+        setProgress(false);
+      }),
+      k8sGet(VolumeSnapshotClassModel, resource.spec.volumeSnapshotClassName),
+    ])
+      .then((obj) => {
+        setPVCData(obj?.[0]);
+        setDriver(obj?.[1]?.driver);
+        setProgress(false);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching objects', e); // Not important to show user
+        setProgress(false);
+      });
+  }, [namespace, resource]);
+
+  const onlyPvcSCs = (scObj: StorageClassResourceKind) => {
+    if (driver) {
+      return scObj.provisioner.includes(driver);
+    }
+    return false;
+  };
+
+  const requestedSizeInputChange = ({ value, unit }) => {
+    setRequestedSize(value);
+    setRequestedUnit(unit);
+  };
+
+  const handleStorageClass = (updatedStorageClass) => {
+    const scName: string = updatedStorageClass?.metadata?.name || '';
+    setPVCStorageClass(scName);
+  };
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const snapshotName = getName(resource);
+    const accessModes = pvcData?.spec?.accessModes || ['ReadWriteOnce'];
+    const restorePVCTemplate: K8sResourceKind = {
+      apiVersion: PersistentVolumeClaimModel.apiVersion,
+      kind: PersistentVolumeClaimModel.kind,
+      metadata: {
+        name: restorePVCName,
+        namespace,
+      },
+      spec: {
+        storageClassName: pvcSC,
+        dataSource: {
+          name: snapshotName,
+          kind: VolumeSnapshotModel.kind,
+          apiGroup: VolumeSnapshotModel.apiGroup,
+        },
+        accessModes,
+        resources: {
+          requests: {
+            storage: `${requestedSize}${requestedUnit}`,
+          },
+        },
+      },
+    };
+
+    // eslint-disable-next-line promise/catch-or-return
+    handlePromise(k8sCreate(PersistentVolumeClaimModel, restorePVCTemplate, namespace)).then(
+      (newPVC) => {
+        close();
+        history.push(resourceObjPath(newPVC, referenceFor(newPVC)));
+      },
+    );
+  };
+
+  return (
+    <div className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>Restore as new PVC</ModalTitle>
+      <Form onSubmit={submit} name="form">
+        <ModalBody>
+          <p>When restore action is finished a new crash-consistent PVC copy will be created.</p>
```

Comment:
This should have a snapshot name to tell which snapshot user is trying to restore.

Reply:
ack